### PR TITLE
Display available keyboard shortcuts in context menus

### LINKED ISSUE #873: With "Show extensions for known file types" disabled, when renaming a file you need to type its extension or it will be unformatted
bug

### LINKED ISSUE #1492: Keyboard shortcuts on context menus
**Is your feature request related to a problem? Please describe.**
Keyboard shortcuts on context menus

**Describe the solution you'd like**
after pressing the menu button on keyboard, i would like to have keyboard shortcuts to use most of the items listed

**Describe alternatives you've considered**
Using shortcuts for copy, cut, paste is available without using the context menus, however, certain functions would require the context menu usage, such as, open with code, properties, etc

**Additional context**
Not all list items in the context menu would need the shortcut

### PR DESCRIPTION
**Pull request by Thomas1664**
Fixes #1492
Fixes #873

### CODE REVIEW

**Comment by yaichenbaum** [comment]
Can you check if this causes the event to be fired twice?

File: Files/Views/LayoutModes/GenericFileBrowser.xaml [hunk 1]
```diff
@@ -133,6 +133,10 @@
                 <MenuFlyoutItem.Icon>
                     <FontIcon FontFamily="{StaticResource FluentUIGlyphs}" Glyph="&#xe916;" />
                 </MenuFlyoutItem.Icon>
+                <MenuFlyoutItem.KeyboardAccelerators>
```
Comment (tsvietOK) [inline Files/Views/LayoutModes/GenericFileBrowser.xaml:136]: This event fired twice when pressing win+r. Can you fix that and check for this issue another events?

**Review by tsvietOK — CHANGES_REQUESTED** [review]

File: Files/Views/LayoutModes/GenericFileBrowser.xaml (hunk 1 above)
Comment (tsvietOK) [reply Files/Views/LayoutModes/GenericFileBrowser.xaml:136]: image

Comment (yaichenbaum) [reply Files/Views/LayoutModes/GenericFileBrowser.xaml:136]: @Thomas1664

Comment (Thomas1664) [reply Files/Views/LayoutModes/GenericFileBrowser.xaml:136]: Sorry for my late reply but I had many things to do in the last weeks.
The layout should be updated only once now.

**Review by tsvietOK — CHANGES_REQUESTED** [review]
Bug: Can't rename file in GridView mode.
Steps:
1. Select file and press F2.
2. Click on empty space or press Enter.

**Comment by Thomas1664** [comment]
@tsvietOK This issue occurs on master as well.
More precise steps to reproduce:

1. Switch to grid view
2. Select an item
3. Press F2 to rename
4. Rename the item
5. Switch to tiles view
6. Rename the same item again by selecting it and pressing F2
7. Switch back to grid view
8. Select the same item
9. Press F2 to rename
10. See error

**Comment by tsvietOK** [comment]
Can't reproduce your issue.

**Comment by Thomas1664** [comment]
@tsvietOK If you referred to a dialog box popping up saying that you can't have 2 files with the same name, this appears in master as well.

**Comment by tsvietOK** [comment]
No, i can't reproduce your issue at all. I am able to rename files without any problem on master branch.

**Comment by tsvietOK** [comment]
Issue i mentioned before still exists with your changes.

**Comment by Thomas1664** [comment]
@tsvietOK And I can't reproduce any issue you have.

**Comment by Thomas1664** [comment]
@tsvietOK This commit might fix your issue (at least it fixes mine)

**Comment by tsvietOK** [comment]
Issue reported by me still present.

**Comment by Thomas1664** [comment]
I still have no crash. How should we continue since I can't reproduce your crash after renaming a file?

**Comment by tsvietOK** [comment]
bug

**Comment by tsvietOK** [comment]
image
That is because of double event call.

**Comment by coheneli98** [comment]
Hey, how are you? I a new at this whole coding business. . . But was wondering, is there a way to set "Files - preview" as the default file explorer? And how can I change the icon, as I was able to do it with all my other apps by using properties but it doesn't seem to work with yours. . . Thanks so much and keep it up!
--

**Comment by Thomas1664** [comment]
> ## Hey, how are you? I a new at this whole coding business. . . But was wondering, is there a way to set "Files - preview" as the default file explorer? And how can I change the icon, as I was able to do it with all my other apps by using properties but it doesn't seem to work with yours. . . Thanks so much and keep it up!

@coheneli98 Please don't ask questions like this in PRs. File an issue or use Discord instead!

**Comment by tsvietOK** [comment]
Can't delete file.
Steps:
1. Open GridView mode
2. Select file
3. Press Delete button
4. Crash

**Comment by tsvietOK** [comment]
Now Delete button is not working in ListView mode/

**Comment by Thomas1664** [comment]
@tsvietOK I can't reproduce a bug with the delete item in context menu. A screenshot would be helpful.

**Comment by tsvietOK** [comment]
I am talking about Delete button on your keyboard.

**Comment by tsvietOK** [comment]
When paste file it shows several copies of these files(PasteItem() called twice), but it fixes when pressing F5.

File: Files/Interacts/Interaction.cs [hunk 2]
```diff
@@ -847,6 +847,9 @@ public async Task<bool> RenameFileItemAsync(ListedItem item, string oldName, str
                 }
                 else
                 {
+                    if (!AppSettings.ShowFileExtensions)
```
Comment (tsvietOK) [inline Files/Interacts/Interaction.cs:850]: Add braces, please.

**Comment by Thomas1664** [comment]
@tsvietOK Please check if there are any other issues in this PR so I can fix them all at once.

**Comment by tsvietOK** [comment]
@Thomas1664 Sure

**Comment by tsvietOK** [comment]
Cut item called twice.

**Comment by tsvietOK** [comment]
"Copy item" called twice.

File: Files/Views/LayoutModes/GridViewBrowser.xaml [hunk 3]
```diff
@@ -158,6 +164,9 @@
                     <MenuFlyoutItem.Icon>
                         <FontIcon FontFamily="{StaticResource FluentUIGlyphs}" Glyph="&#xea55;" />
                     </MenuFlyoutItem.Icon>
+                    <MenuFlyoutItem.KeyboardAccelerators>
```
Comment (tsvietOK) [inline Files/Views/LayoutModes/GridViewBrowser.xaml:167]: Fix tag padding, please.

Comment (tsvietOK) [reply Files/Views/LayoutModes/GridViewBrowser.xaml:167]: Seems like all bugs are fixed, except this padding.

**Review by tsvietOK — APPROVED** [review]

File: Files/Views/LayoutModes/GridViewBrowser.xaml [hunk 4]
```diff
@@ -250,6 +259,9 @@
                 <MenuFlyoutItem.Icon>
                     <FontIcon FontFamily="{StaticResource FluentUIGlyphs}" Glyph="&#xe916;" />
                 </MenuFlyoutItem.Icon>
+                <MenuFlyoutItem.KeyboardAccelerators>
+                    <KeyboardAccelerator Key="F5" />
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GridViewBrowser.xaml:263]:
```suggestion
                    <KeyboardAccelerator Key="F5" IsEnabled="False" />
```

File: Files/Views/LayoutModes/GridViewBrowser.xaml [hunk 5]
```diff
@@ -158,6 +164,9 @@
                     <MenuFlyoutItem.Icon>
                         <FontIcon FontFamily="{StaticResource FluentUIGlyphs}" Glyph="&#xea55;" />
                     </MenuFlyoutItem.Icon>
+                    <MenuFlyoutItem.KeyboardAccelerators>
+                        <KeyboardAccelerator Modifiers="Control,Shift" Key="N" />
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GridViewBrowser.xaml:168]:
```suggestion
                        <KeyboardAccelerator Modifiers="Control,Shift" Key="N" IsEnabled="False" />
```

File: Files/UserControls/ModernNavigationToolbar.xaml [hunk 6]
```diff
@@ -1008,6 +1008,9 @@
                             <MenuFlyoutItem.Icon>
                                 <FontIcon FontFamily="{StaticResource CustomGlyph}" Glyph="&#xF104;" />
                             </MenuFlyoutItem.Icon>
+                            <MenuFlyoutItem.KeyboardAccelerators>
+                                <KeyboardAccelerator Modifiers="Control" Key="T" />
```
Comment (yaichenbaum) [inline Files/UserControls/ModernNavigationToolbar.xaml:1012]:
```suggestion
                                <KeyboardAccelerator Modifiers="Control" Key="T" IsEnabled="False" />
```

File: Files/UserControls/ModernNavigationToolbar.xaml [hunk 7]
```diff
@@ -1017,6 +1020,9 @@
                             <MenuFlyoutItem.Icon>
                                 <FontIcon FontFamily="{StaticResource FluentUIGlyphs}" Glyph="&#xE973;" />
                             </MenuFlyoutItem.Icon>
+                            <MenuFlyoutItem.KeyboardAccelerators>
+                                <KeyboardAccelerator Modifiers="Control" Key="N" />
```
Comment (yaichenbaum) [inline Files/UserControls/ModernNavigationToolbar.xaml:1024]:
```suggestion
                                <KeyboardAccelerator Modifiers="Control" Key="N" IsEnabled="False" />
```

File: Files/UserControls/ModernNavigationToolbar.xaml [hunk 8]
```diff
@@ -1039,6 +1045,9 @@
                             <MenuFlyoutItem.Icon>
                                 <FontIcon FontFamily="{StaticResource FluentUIGlyphs}" Glyph="&#xE9B2;" />
                             </MenuFlyoutItem.Icon>
+                            <MenuFlyoutItem.KeyboardAccelerators>
+                                <KeyboardAccelerator Modifiers="Control" Key="V" />
```
Comment (yaichenbaum) [inline Files/UserControls/ModernNavigationToolbar.xaml:1049]:
```suggestion
                                <KeyboardAccelerator Modifiers="Control" Key="V" IsEnabled="False" />
```

File: Files/UserControls/StatusBarControl.xaml [hunk 9]
```diff
@@ -62,6 +62,9 @@
                             <MenuFlyoutItem.Icon>
                                 <FontIcon FontFamily="{StaticResource FluentUIGlyphs}" Glyph="&#xEB54;" />
                             </MenuFlyoutItem.Icon>
+                            <MenuFlyoutItem.KeyboardAccelerators>
+                                <KeyboardAccelerator Modifiers="Control" Key="A" />
```
Comment (yaichenbaum) [inline Files/UserControls/StatusBarControl.xaml:66]:
```suggestion
                                <KeyboardAccelerator Modifiers="Control" Key="A" IsEnabled="False" />
```

File: Files/Views/LayoutModes/GenericFileBrowser.xaml [hunk 10]
```diff
@@ -134,6 +134,9 @@
...
+                    <KeyboardAccelerator Key="F5" />
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GenericFileBrowser.xaml:138]:
```suggestion
                    <KeyboardAccelerator Key="F5" IsEnabled="False" />
```

File: Files/Views/LayoutModes/GenericFileBrowser.xaml [hunk 11]
```diff
@@ -165,6 +171,9 @@
                 <MenuFlyoutSubItem.Icon>
                     <FontIcon FontFamily="{StaticResource FluentUIGlyphs}" Glyph="&#xe903;" />
                 </MenuFlyoutSubItem.Icon>
+                <MenuFlyoutSubItem.KeyboardAccelerators>
+                    <KeyboardAccelerator Modifiers="Control,Shift" Key="N" />
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GenericFileBrowser.xaml:175]:
```suggestion
                    <KeyboardAccelerator Modifiers="Control,Shift" Key="N" IsEnabled="False" />
```

File: Files/Views/LayoutModes/GenericFileBrowser.xaml [hunk 12]
```diff
@@ -277,6 +286,9 @@
                 <MenuFlyoutItem.Icon>
                     <FontIcon FontFamily="{StaticResource FluentUIGlyphs}" Glyph="&#xe916;" />
                 </MenuFlyoutItem.Icon>
+                <MenuFlyoutItem.KeyboardAccelerators>
+                    <KeyboardAccelerator Key="F5" />
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GenericFileBrowser.xaml:290]:
```suggestion
                    <KeyboardAccelerator Key="F5" IsEnabled="False" />
```

File: Files/Views/LayoutModes/GridViewBrowser.xaml [hunk 13]
```diff
@@ -119,6 +119,9 @@
                 <MenuFlyoutItem.Icon>
                     <FontIcon FontFamily="{StaticResource FluentUIGlyphs}" Glyph="&#xe916;" />
                 </MenuFlyoutItem.Icon>
+                <MenuFlyoutItem.KeyboardAccelerators>
+                    <KeyboardAccelerator Key="F5" />
```
Comment (yaichenbaum) [inline Files/Views/LayoutModes/GridViewBrowser.xaml:123]:
```suggestion
                    <KeyboardAccelerator Key="F5" IsEnabled="False" />
```

File: Files/Views/ModernShellPage.xaml.cs [hunk 14]
```diff
@@ -983,7 +983,7 @@ private async void KeyboardAccelerator_Invoked(KeyboardAccelerator sender, Keybo
             {
                 switch (args.KeyboardAccelerator.Key)
                 {
-                    case VirtualKey.F2: //F2, rename
+                    case VirtualKey.F2: //F2, rename
```
Comment (yaichenbaum) [inline Files/Views/ModernShellPage.xaml.cs:986]:
```suggestion
                    case VirtualKey.F2: //F2, rename
```

**Review by yaichenbaum — CHANGES_REQUESTED** [review]

**Review by yaichenbaum — APPROVED** [review]

**Review by tsvietOK — APPROVED** [review]

**Pull request merged**